src/main/java/edu/unc/lib/boxc/migration/cdm/services/PostMigrationReportService.java

```diff
@@ -175,6 +198,21 @@ private String extractTitle(String cdmId) {
         return null;
     }
 
+    private String getMatchingValue(String cdmId) throws IOException {
+        var sourceFilesInfo = sourceFileService.loadMappings();
```

Comment:
It doesn't look like there's any sort of caching of the results in sourceFileService, so I would suggest updating PostMigrationReportService to cache the response itself and use that in each of these. Otherwise reading the entire source file mapping twice for file entry could cause some slowness in larger projects.